### PR TITLE
add scope to foreach iteration symbols

### PR DESCRIPTION
**Pull request by WalterBright**
spec changes corresponding to https://github.com/dlang/dmd/pull/12912

### CODE REVIEW

File: spec/statement.dd [hunk 1]
```diff
@@ -488,8 +489,10 @@ $(P
         At the start of each iteration, the variables declared by
         the $(I ForeachTypeList)
         are set to be a copy of the elements of the aggregate.
-        If the variable is $(D ref), it is a reference to the
+        If the $(I ForeachTypeAttribute) is $(D ref), it is a reference to the
         contents of that aggregate.
+        If the $(I ForeachTypeAttribute) is $(D scope), the $(I ForeachType) declaration
+        will have $(D scope) semantics.
```
Comment (Geod24) [inline spec/statement.dd:495]: Isn't that redundant ?

Comment (WalterBright) [reply spec/statement.dd:495]: I thought it necessary because of the `ref` explanation.

Comment (Geod24) [reply spec/statement.dd:495]: I think the `ref` is necessary because the previous line mentions a copy.

**Review by Geod24 — APPROVED** [review]

**Comment by nordlow** [comment]
Awesome!

**Pull request merged**